api/objects/validate

```diff
@@ -79,18 +79,29 @@ case "update-service":
     }
 
     if($data['Ports']) {
+        $portValidator = $v->createValidator(Validate::PORTNUMBER);
         if (strpos($data['Ports'][0], "-") !== false ) { # port range
-            $tmp = explode("-",$data['Ports'][0]);
-            # two non-zero integers, right number must be greater then left one
-            if ( !isset($tmp[0], $tmp[1]) || !(int)$tmp[0] || !(int)$tmp[1] || (int)$tmp[0] >= (int)$tmp[1] ) {
-                $v->addValidationError('Ports', 'not_valid_port_range');
+            # Verify that we use either a comma separated list OR a range of ports
+            if (isset ($data['Ports'][1])) {
+                $v->addValidationError('Ports', 'must_be_port_list_or_a_port_range');
+            } else {
+                $tmp = explode("-",$data['Ports'][0]);
+                # Verify that each port range is a good port value
+                foreach( $tmp as $port) {
+                    if(!$portValidator->evaluate($port)) {
+                        $v->addValidationError('Ports', 'not_valid_port_range', $data['Ports'][0]);
```

Comment:
Do not evaluate further elements if one fails

```suggestion
                        $v->addValidationError('Ports', 'not_valid_port_range', $data['Ports'][0]);
                        break;
```

BTW the _foreach_ loop could be avoided: just repeat the same evaluation for `$tmp[0]` and `$tmp[1]` -- or `$lower_port` and `$upper_port` ;)

Reply:
done